### ISSUE TITLE
Clarification to the "factor" parameter

### ISSUE THREAD

**Issue by neon5d** [MengyuanWang524/Lasso#1]
Hello,
Thank you for sharing your code. 
I have three questions.
I can see the "factor" parameter at "onionRecn(lambda, factor)" and "wedgeRecn".
1. I can't find any description of the "factor" parameter in your paper.
Which parameter should I refer to for the "factor" parameter in your paper?
2. The "factor" parameter at "onionRecn(lambda, factor)" and "wedgeRecn".
Are these the same parameter?
If it is, then I wondered why it is applied differently.
At onionRecn(lambda, factor),
The "factor" parameter only affects "dzRecn = dzFFT / factor;"
However, at "wedgeRecn",
The "factor" parameter affects both "dzRecn = dzFFT / factor;" and "numRecn = 2048 * factor / 2;".
Can you clarify these?
3. img is undefined at 'wedgeRecn'.
Can you fix this issue?

Thank you

**Comment by MengyuanWang524** [comment]
Hello,
Thank you for your interests in our work, 


In our paper, we analyzed how IFFT limit the resolution and gave two girds.
The FFT reconstruction grid has a grid size which is determined by the bandwidth(See code 'onionRecn' Line28)
Benefiting from the optimization we could have a user-defined reconstruction grid:
1. Here 'factor' means the resolution improvement that how much the reconstructed pixel size could be set smaller than the FFT grid.
 2. The imaging depth could also be self-defined, we use '/2' here to have a range of two times smaller, you could choose those parameters
in your case for a better reconstruction.


Thanks again for your attention again! We haven't perfected the code for a final version, any question is welcomed!


Best,


Mengyuan Wang.
 

















At 2020-06-28 13:31:24, "neon5d" <notifications@github.com> wrote:

Hello,
Thank you for sharing your code.
I have one question.
I can see the "factor" parameter at "onionRecn(lambda, factor)" and "wedgeRecn".

I can't find any description of the "factor" parameter in your paper.
Which parameter should I refer to for the "factor" parameter in your paper?
The "factor" parameter at "onionRecn(lambda, factor)" and "wedgeRecn".
Are these the same parameter?
If it is, then I wondered why it is applied differently.
At onionRecn(lambda, factor),
The "factor" parameter only affects "dzRecn = dzFFT / factor;"
However, at "wedgeRecn",
The "factor" parameter affects both "dzRecn = dzFFT / factor;" and "numRecn = 2048 * factor / 2;".

Can you clarify these?
Thank you

**Comment by neon5d** [comment]
Thank you for your comments.
I am still confused.
1. In practice, the pitch size of a linear image sensor (CCD image sensor in GAN620C1 Thorlabs) is generally larger than 5um (I checked Sony and Hamamatsu (7um or 14um)). 
In the paper, \delta_k of Figure 1 is larger than 5um (the pitch size of an image sensor).
From the Matlab code, dzFFT is 1.9um. 
I am very confused about how I take into account the pitch size limitation in practice.

From the Matlab code, fringe = h5read('rawSpectrumOnionThorlab.h5','/rawData');
The fringe was already sampled by \delta_k with the image sensor pitch.
Mathematically, everything is clear, but how can I consider the practical matter (the pitch size limitation) into your proposed method?

2. img is undefined at 'wedgeRecn'.
Can you fix this issue?

Thank you,

**Comment by ghost** [comment]
Hi there,

1.	If you are referring to the Figure 1 in the manuscript (https://arxiv.org/abs/2001.03129), the \Delta k (capital delta) refers to the entire wavenumber range sensed by the CCD, i.e. the wavenumber range related to [791.6, 994] nm. The \delta_k (small delta  denotes the sampling interval in the wavenumber domain, which is calculated by dividing the wavenumber range (capital Delta k) by the total pixel number 2048. Both \delta_k and \Delta k have a unit of m^{-1}.

In the current study, we do not take the pixel response function into consideration; each pixel is simply modeled as a single sampling point. Therefore, it doesn’t really matter how large the pitch is.

On the other hand, dzFFT defines the corresponding pixel size in the reconstruction (or object or spatial you name it) domain. It is also known as the “digital resolution” for OCT image. It is defined by \Delta k through the Fourier relationship.

To sum up, we don’t take the pixel response function into consideration in our study. I think a better modelling might want to include that as well, but I don’t know how much improvement it could bring.

2.	The bug is fixed now. Thanks for pointing out.

Best,

Yuye

**Comment by neon5d** [comment]
Thank you for your detailed explanation and fixed the bug.
Thanks a lot.
